### ISSUE TITLE
Usage visibility qualifiers and group options

### ISSUE THREAD

**Issue by jonathanolson** [phetsims/area-model-common#139]
It was recommended to use `// @public (a11y)` in this sim (for things that are only public because of a11y needs), and `// @public (phet-io)` is also used elsewhere.

If this is common and helpful, let's add it to the code-review checklist. If anyone has objections, note them here? (I have a gut instinct that this is a code smell meaning either "they should be private and we should figure a way of not breaking that for phet-io/a11y", or "it's fine for these to be public, but we only did so because of X").

**Comment by pixelzoom** [comment]
Related, and I'm sure where to create an issue, so I'll tack it on here... I'd like to see options grouped similarly, e.g.:

```js
options = _.extends( {

  fill: ‘white’,
  stroke: ‘black’,
  …

  // phet-io
  tandem: tandem,
  …

  // a11y 
  tagName: 'p',
  descriptionContent: 'blah blah blah',
  …

}, options );
```

It looks like this is the general convention that is being followed in most places.  But given the proliferation of phet-io and a11y options, I think it would be useful to formalize this convention.

**Comment by zepumph** [comment]
Agreed with @pixelzoom greatly. Especially for a11y, which has upwards of 20 options, I think grouping and commenting is a must.

**Comment by pixelzoom** [comment]
Or how about using nested options to group a11y and phet-io options?  That seems like a more reliable grouping approach.  E.g.:

```js
options = _.extends( {

  fill: ‘white’,
  stroke: ‘black’,
  …

  phetio: {
    tandem: tandem
    ...
  }
  …

  a11y: { 
    tagName: 'p',
    descriptionContent: 'blah blah blah',
    …
  }

}, options );
```

**Comment by jonathanolson** [comment]
> If we use the type name as the grouping key

That sounds super inconvenient.

Also, things are getting quite verbose having to tag each section of options. Are we sure that this has a lot of benefits?

**Comment by pixelzoom** [comment]
I don't see a lot of benefits to grouping options as in https://github.com/phetsims/area-model-common/issues/139#issuecomment-389297347.   But I do think we should discuss the pros/cons of grouping options that are related to a11y and phetio.

**Comment by jbphet** [comment]
I like the idea of grouping a11y and phet-io options, since these are generally quite distinct from other options.  Further grouping seems potentially cumbersome IMO, at least until some new pervasive feature comes along.

**Comment by jonathanolson** [comment]
> I like the idea of grouping a11y and phet-io options, since these are generally quite distinct from other options.

I'm interested then in what exactly are we grouping? Do we group:

- A title `{string}` (option for a component) that is only used for a11y?
- A title `{string}` (option for a component) that is used to create a Text node AND for a11y?

Or are we talking about ONLY grouping options specific to the Accessibility trait? (ACCESSIBILITY_OPTION_KEYS in Accessibility.js)

**Comment by samreid** [comment]
I agree with @jonathanolson's questions, and also want to clarify if we are discussing grouping structurally, or if it would be preferable to use code commenting to group the options.

**Comment by jessegreenberg** [comment]
Sounds like we are thinking of using a new pattern thing for a11y/phet-io, and the current pattern for all others (like those mixed in with Paintable). I am not sure why yet. Accessibility.js uses scenery/Node's mutate pattern, so I would still want the setters available
```js
options = _.extends( {
  // ...
  Accessibility: {
    tagName: 'p',
    descriptionContent: 'blah blah blah'
  }

  // ...
}, options );
```

I would still want 
```js
// update tag name and description
this.tagName = 'div';
this.descriptionContent = 'Some new description'
```

**Comment by samreid** [comment]
@jonathanolson did we address the initial comment in this issue?

**Comment by jonathanolson** [comment]
Well enough in my opinion (that's it's a helpful way to document why something is public).
